python/ray/rllib/rollout.py

```diff
@@ -122,16 +121,40 @@ def rollout(agent, env_name, num_steps, out=None, no_render=True):
         if out is not None:
             rollout = []
         state = env.reset()
+        if multiagent:
+            agents = state.keys()
         done = False
         reward_total = 0.0
         while not done and steps < (num_steps or steps + 1):
-            if use_lstm:
-                action, state_init, logits = agent.compute_action(
-                    state, state=state_init)
+            if multiagent:
+                action_dict = {}
+                for agent_id in agents:
```

Comment:
Ah you should iterate over state.keys() instead, since new agents might appear in the env.